src/kOS/Suffixed/Part/EngineValue.cs

```diff
@@ -213,6 +246,215 @@ public ScalarValue GetPossibleThrustAtAtm(ScalarValue atmPressure)
+                }
+            }
+
+            return massFlow;
+        }
+
+        public BooleanValue GetUllage()
+        {
+            bool ullage = false;
+
+            if (HasRealFuels)
+            {
+                foreach (ModuleEngines e in FilteredEngineList)
+                {
+                    var ullageField = e.GetType().GetField("ullage");
+                    if (ullageField != null)
+                    {
+                        ullage |= (bool)ullageField.GetValue(e);
+                    }
+                }
+            }
+
+            return ullage;
+        }
+
+        public ScalarValue GetFuelStability()
+        {
+            // Default to full stability
+            float stability = 1.0f;
+
+            if (HasRealFuels)
+            {
+                foreach (ModuleEngines e in FilteredEngineList)
+                {
+                    var ullageField = e.GetType().GetField("ullage");
+                    var pressureFedField = e.GetType().GetField("pressureFed");
+                    var ullageSetField = e.GetType().GetField("ullageSet");
+
+                    if (ullageField != null && pressureFedField != null && ullageSetField != null)
+                    {
+                        bool pressureOK = !(bool)pressureFedField.GetValue(e);
+                        if (!pressureOK)
+                        {
+                            var pressureOKMethod = ullageSetField.GetType().GetMethod("PressureOK", BindingFlags.Public | BindingFlags.Instance);
+                            pressureOK = pressureOKMethod != null ? (bool)pressureOKMethod.Invoke(ullageSetField.GetValue(e), new object[] { }) : true;
+                        }
+
+                        if (!pressureOK)
+                        {
+                            // No feed pressure
+                            stability = 0;
+                        }
+                        else if ((bool)ullageField.GetValue(e))
+                        {
+                            // Use minimum value if multiple engines
+                            var ullageStabilityMethod = ullageSetField.GetType().GetMethod("GetUllageStability", BindingFlags.Public | BindingFlags.Instance);
+                            stability = Mathf.Min(stability, ullageStabilityMethod != null ? (float)ullageStabilityMethod.Invoke(ullageSetField.GetValue(e), new object[] { }) : 1.0f);
+                        }
+                    }
+                }
+            }
+
+            return stability;
+        }
+
+        public BooleanValue GetPressureFed()
+        {
+            bool pressureFed = false;
+
+            if (HasRealFuels)
+            {
+                foreach (ModuleEngines e in FilteredEngineList)
+                {
+                    var pressureFedField = e.GetType().GetField("pressureFed");
+                    if (pressureFedField != null)
+                    {
+                        pressureFed |= (bool)pressureFedField.GetValue(e);
```

Comment:
(same comment as above, as this is the same pattern - using `|=` to affect a boolean technically works but feels so very wrong to me.  Also, why not break the foreach loop once its true, if that's the goal of this.